### PR TITLE
1.0 develop

### PR DESCRIPTION
**Pull request by grawk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
...
+$ BROWSER=chrome node examples/setupWithConfigDir.js
+Nemo successfully launched chrome
+```
+
+Here, confit resolves the `BROWSER` environment variable and overrides `BROWSER` from config.json
+
+What if we set both?
+
+```bash
+$ BROWSER=chrome node examples/setupWithConfigDir.js BROWSER=phantomjs
```
Comment (nikulkarni) [inline README.md]: last argument should be `--BROWSER=phantomjs`. `--` is missing

File: README.md [hunk 2]
```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
...
+This configuration object is optional, as long as you've got `nemoData` set as an environment variable (see below).
 
-Used as a root location for finding 'locator' files in your test suite. Also can be required by other plugins
-which need to require modules from your test suite.
+### driver
 
-### targetBrowser
+Here are the `driver` properties recognized by Nemo. This is ALL of them. Please be aware that you really only need to supply "browser" to get things working initially.
+
+#### browser (optional)
```
Comment (nikulkarni) [inline README.md]: It looks like browser is not optional. we don't default it to anything and that makes driver not initialize. For example below example doesn't work locally 
```
var nemo = Nemo({
    "driver": {
    },
    'data': {
        'baseUrl': 'https://www.paypal.com'
    }
}, function (err) {
    //always check for errors!
    if (!!err) {
        console.log('Error during Nemo setup', err);
    }
    nemo.driver.get(nemo.data.baseUrl);
    nemo.driver.getCapabilities().
        then(function (caps) {
            console.info("Nemo successfully launched-", caps.caps_.browserName);
        });
    nemo.driver.quit();
});```

File: README.md [hunk 3]
```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
...
+Set local to true if you want Nemo to attempt to start a standalone binary on your system (like selenium-standalone-server) or use a local browser/driver like Chrome/chromedriver or PhantomJS.
 
-### targetServer (optional)
+
+#### server (optional)
 
 Webdriver server URL you wish to use.
 
-If you are using sauce labs, make sure `targetServer` is set to correct url like `"http://yourkey:yoursecret@ondemand.saucelabs.com:80/wd/hub"`
+If you are using sauce labs, make sure `server` is set to correct url like `"http://yourkey:yoursecret@ondemand.saucelabs.com:80/wd/hub"`
```
Comment (nikulkarni) [inline README.md]: Only setting server with correct sauce url is not enough. Users also must pass `username` and `accessKey` as driver capabilities. Without out it test fails. I tested this and we need to specify this in README

"serverCaps" : {
            "username" : "yourusername",
            "accessKey" : "yourkey"
        }

File: README.md [hunk 4]
```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
...
-Browser you wish to automate. Make sure that your chosen webdriver has this browser option available
+#### browser (optional)
 
-### localServer (optional, defaults to false)
+Browser you wish to automate. Make sure that your chosen webdriver has this browser option available. While this is "optional" you must choose a browser. Either use this property or the "builders.forBrowser" option (see below).
 
-Set localServer to true if you want Nemo to attempt to start a standalone binary on your system (like selenium-standalone-server) or use a local browser/driver like Chrome/chromedriver or PhantomJS.
+#### local (optional, defaults to false)
 
+Set local to true if you want Nemo to attempt to start a standalone binary on your system (like selenium-standalone-server) or use a local browser/driver like Chrome/chromedriver or PhantomJS.
```
Comment (nikulkarni) [inline README.md]: If server property is absent, run is treated as local

File: package.json [hunk 5]
```diff
@@ -1,30 +1,29 @@
...
-    "xunit-file": "~0.0.4",
+    "async": "~0.2.8",
+    "confit": "^2.0.0",
     "debug": "^2.1.0",
-    "async": "0.1.x",
+    "lodash": "^2.4.0",
     "nconf": "~0.6.7",
-    "lodash": "^2.4.0"
+    "path": "^0.11.14",
+    "selenium-webdriver": "~2.44.0",
```
Comment (nikulkarni) [inline package.json]: can we bump it to 2.45.0 along with this?

Comment (grawk) [inline package.json]: Unfortunately we cannot. 2.45.1 will not launch either Chrome or Firefox on Mac OS. We will have to stay glued to 2.44 until they resolve that issue.

**Pull request merged**